File: solutions/ruby/armstrong-numbers/1/armstrong_numbers.rb
module ArmstrongNumbers
  def self.include?(number)
    digits = number.to_s.chars.map { |digit| digit.to_i }
    num_of_digits = digits.length
    sum_of_digits_raised_to_num_of_digits = digits.map { |digit| digit ** num_of_digits }.sum
    sum_of_digits_raised_to_num_of_digits == number
  end
end

File: solutions/cpp/armstrong-numbers/1/armstrong_numbers.cpp
#include "armstrong_numbers.h"

namespace armstrong_numbers {

bool is_armstrong_number(unsigned int number) {
    auto number_of_digits = number == 0? 1: int(log10(number)) + 1;
    unsigned int sum_of_powers = 0;
    auto test = number;
    while (test != 0) {
        auto digit = test % 10;
        test /= 10;
        sum_of_powers += pow(digit, number_of_digits);
    }
    return (number == sum_of_powers);
}

}  // namespace armstrong_numbers
